file_management/include/file_management/file_upload/file_manager.h

```diff
@@ -69,6 +69,10 @@ class DataReader : public Service {
     const FileObject<T> &log_messages) = 0;
 
   virtual void setStatusMonitor(std::shared_ptr<StatusMonitor> status_monitor) = 0;
+
+  virtual void discardFiles(){
```

Comment:
Why is this a no-op?

Reply:
I've moved the implementation of this function up to the FileManager class. I think it makes sense to allow FileManager to utilize this in the future but if that's not appropriate please let me know.

The MockDataReader function in file_upload_streamer_test will not implement this because we are testing for the logic in log_batch_test. Please let me know if that's not correct.